grammar/grammar.y

```diff
@@ -332,21 +334,49 @@ ExprList <- (Expr COMMA)* Expr?
 
 # *** Tokens ***
 eof <- !.
+bin <- [01]
+bin_ <- '_'? bin
+oct <- [0-7]
+oct_ <- '_'? oct
 hex <- [0-9a-fA-F]
-hex_ <- ('_'/hex)
+hex_ <- '_'? hex
 dec <- [0-9]
-dec_ <- ('_'/dec)
-
-dec_int <- dec (dec_* dec)?
-hex_int <- hex (hex_* dec)?
+dec_ <- '_'? dec
+
+bin_int <- bin bin_*
+oct_int <- oct oct_*
+dec_int <- dec dec_*
+hex_int <- hex hex_*
+
+utf8_trailer <- [�-�]
+oxf4 <- [�-�]
+oxf1_oxf3 <- [�-�]
+oxf0 <- [�-�]
+ox90_0xbf <- [�-�]
+oxee_oxef <- [�-�]
+oxed <- [�-�]
+ox80_ox9f <- [�-�]
+oxe1_oxec <- [�-�]
+oxe0 <- [�-�]
+oxa0_oxbf <- [�-�]
+oxc2_oxdf <- [�-�]
```

Comment:
This is a good change, but is it possible to express these with escape sequences or hex literals so we don't need to require a special encoding for this file?

Reply:
As far as I can tell peg/leg supports octal escapes but not hex. I replaced the latin1 chars with octal escapes so ASCII/utf8 is fine for this file again.